src/plugins/intel_gpu/src/graph/include/reshape_inst.h

```diff
@@ -63,7 +63,7 @@ struct typed_program_node<reshape> : public typed_program_node_base<reshape> {
     }
 
     void adjust_output_padding() {
-        if (!is_in_place() || !has_padding())
```

Comment:
Basically check "!is_in_place()" condition has conflict behind "!has_padding()", Remove "!is_in_place()" is needed, but this cause unexpected adjust_output_padding() call in reshape can't be optimized out case. Is it expected? Ex> reshape has fused_primitives.

Reply:
A proper usage of this logic is to adjust outer padding as long as it has (outer_padding && no_spatial_padding).
How about adding exception about 'has_fused_primitives' to 'has_outer_padding_offset' and changing the name to 'has_optimizable_outer_padding_offset'?

Reply:
Discuss the offline, We aligned adjust_output_padding() will be only called when Reshape can be optimized out. And this changes will be updated with iGPU bug fix PR in the future.